website_sale_checkout_country_vat/views/templates.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- © 2016 Sergio Teruel <sergio.teruel@tecnativa.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl-3). -->
+<openerp>
+<data>
+
+<template id="country_dropdown"
+          inherit_id="website_snippet_country_dropdown.country_dropdown"
+          primary="True">
+
+    <xpath expr="//t[@t-set='default_country']" position="attributes">
+        <attribute name="t-value">website.vat_code_id</attribute>
+    </xpath>
+
+    <xpath expr="//input[@id='no_country_field']" position="attributes">
+        <attribute name="t-att-value">checkout.get('vat')</attribute>
```

Comment:
Oh sorry, I missed that you were using `primary=True`. Anyway, the variables approach is much better.